R/capture-condition.R

```diff
@@ -1,3 +1,25 @@
+
+bang <- function(expr, env = caller_env()) {
+  eval_bare(enexpr(expr), env)
+}
+new_capture <- function(class) {
+  exiting_handlers <- rep_named(class, list(identity))
+
+  calling_handlers <- rep_named(class, alist(function(cnd) {
+    cnd <- cnd_entrace(cnd)
+    return_from(env, cnd)
+  }))
+
+  bang(function(code, entrace = FALSE) {
```

Comment:
I think this might be simpler with `new_function()`, since I think the environment of the generated function should be the package env, not an execution environment of `new_capture()`.